src/lib/stateBasedTestHelper.ts

```diff
@@ -115,24 +113,14 @@ interface SagaTestBuilder4 {
   forReducer: (reducer: Function) => Promise<void>;
 }
 
-interface SagaEnv<StateT, ActionT> {
-  call: <T extends (...args: any[]) => any>(f: T) => ReturnType<T>;
-  select: <R>(s: Selector<StateT, R>) => R;
-  dispatch: (action: ActionT) => void;
-}
-
 export async function testSagaWithState<StateT, Payload>(
-  saga: Saga<StateT, any, Payload>,
-  initialAction: Action<Payload>,
+  saga: Saga<StateT, Payload>,
+  initialPayload: Payload,
```

Comment:
hmm, seems like my comment got los.

I wrote that I used `Action<Payload>` on purpose here, because I think we should have the test helper first send the message to the reducer, to more closely model the real-world behavior.

Reply:
I would agree that the reducer should get the action first, but I wouldn't require to test a saga with a reducer.

Also, we only pass in the payload in the saga, so it's similar to the saga here.

Reply:
But then we can't give it to the reducer first / the tester would have to manually create the initial state and then call the reducer with the "previous initial state" and the action.

So if one would do that properly, it would be way more verbose than just providing the `Action` here.

Reply:
We can create the action with the action creator from the saga and call the reducer with that.